classes/plugins/event.py

```diff
@@ -0,0 +1,56 @@
+import os
+import builtins
+import logging
+import pickle
+import typing
+_pickleLevel_ = 5 #pickle level to use when comprssing and storing events
+def _save_(events):
+    folder = os.path.dirname(os.path.realpath(__file__))
+    with open(f'{folder}/events.pickle',"wb") as file:
+        pickle.dump(events,file,protocol=_pickleLevel_)
+
+def _load_():
+    folder = os.path.dirname(os.path.realpath(__file__))
+    try:
+        with open(f'{folder}/events.pickle',"rb") as file:
+            return pickle.load(file)
+    except FileNotFoundError:
+        return {}
+def _reset_():
+    folder = os.path.dirname(os.path.realpath(__file__))
+    with open(f'{folder}/events.pickle',"wb") as file:
+        pickle.dump({},file,protocol=_pickleLevel_)
+
+_events_ = {}
+
+def registerEvent(name: str):
+    _events_ = _load_()
```

Comment:
we are reassigning a higher scoped variable each time it is run, is that intended?